cruise-control-metrics-reporter/src/main/java/com/linkedin/kafka/cruisecontrol/metricsreporter/CruiseControlMetricsReporter.java

```diff
@@ -332,10 +335,30 @@ public void onCompletion(RecordMetadata recordMetadata, Exception e) {
     });
   }
 
+  /**
+   * Starting with Kafka 2.6.0, a new class KafkaYammerMetrics provides the default Metrics Registry. The old default
+   * registry does not work with 2.6+. Therfore if the new class exists, we use it and if it doesn't exist we use the
+   * old one. More details can be found here: https://github.com/apache/kafka/blob/2.6.0/core/src/main/java/kafka/metrics/KafkaYammerMetrics.java
+   *
+   * Once CC supports only 2.6.0 and newer, we can clean this up and use only KafkaYammerMetrics all the time.
+   *
+   * @return  MetricsRegistry with Kafka metrics
+   */
+  private MetricsRegistry metricsRegistry()   {
+    try {
+      Class.forName("kafka.metrics.KafkaYammerMetrics");
```

Comment:
This method seems to be called in each cycle of metrics reporting. Can we call the metricsRegistry() method once, cache the regustry and reuse it each time is required?

Reply:
Good point, I added a field and instantiated it in the `init()` method.